helpers/setup-spark-k8s.py

```diff
@@ -91,44 +92,94 @@ def get_defaults_from_kubeconfig(kubeconfig: str, context: str = None) -> Dict:
 
     return defaults
 
-def set_up_user(username: str, name_space: str, defaults: Dict) -> None:
+def set_up_user(username: str, name_space: str, defaults: Dict, mark_primary: bool) -> None:
     namespace = name_space or defaults['namespace']
     kubeconfig = defaults['config']
     context_name = defaults['context']
 
     rolebindingname = username + '-role'
     roleaccess='view'
-    label = 'app.kubernetes.io/managed-by=spark-client'
+    label = utils.get_management_label()
+
     os.system(f"{KUBECTL_CMD} create serviceaccount --kubeconfig={kubeconfig} --context={context_name} {username} --namespace={namespace}")
     os.system(f"{KUBECTL_CMD} create rolebinding --kubeconfig={kubeconfig} --context={context_name} {rolebindingname} --role={roleaccess}  --serviceaccount={namespace}:{username} --namespace={namespace}")
-    os.system(f"{KUBECTL_CMD} label serviceaccount --kubeconfig={kubeconfig} --context={context_name} {username} {label} --namespace={namespace}")
-    os.system(f"{KUBECTL_CMD} label rolebinding --kubeconfig={kubeconfig} --context={context_name} {rolebindingname} {label} --namespace={namespace}")
 
-def setup_spark_conf_defaults(username: str, namespace: str) -> None:
-    DYNAMIC_DEFAULTS_CONF_FILE = utils.get_dynamic_defaults_conf_file()
-    generated_defaults = utils.generate_spark_default_conf()
-    if username:
-        generated_defaults['spark.kubernetes.authenticate.driver.serviceAccountName'] = username
-    if namespace:
-        generated_defaults['spark.kubernetes.namespace'] = namespace
+    primary_label_to_remove = utils.get_primary_label(label=False)
+    primary_label_full = utils.get_primary_label()
+
+    primary = utils.retrieve_primary_service_account_details()
+    is_primary_defined = len(primary.keys()) > 0
+
+    logging.debug(f"is_primary_defined={is_primary_defined}")
+    logging.debug(f"mark_primary={mark_primary}")
+
+    if is_primary_defined and mark_primary:
+        sa_to_unlabel = primary['spark.kubernetes.authenticate.driver.serviceAccountName']
+        namespace_of_sa_to_unlabel = primary['spark.kubernetes.namespace']
+        rolebindingname_to_unlabel = sa_to_unlabel + '-role'
+
+        os.system(f"{KUBECTL_CMD} label serviceaccount --kubeconfig={kubeconfig} --context={context_name} --namespace={namespace_of_sa_to_unlabel} {sa_to_unlabel} {primary_label_to_remove}-")
```

Comment:
Sometimes we use KUBECTL_CMD and sometimes we have a helpers function get_kubectl_cmd, or build_kubectl_cmd. 

When do we use one or another? Do we have use-cases for having multiple ways to do similar things? Can we think of refactors that would allow us to centralize behaviour and implementation, in order to promove code reusability and consistent API/UX?